addons/google_calendar/models/res_users.py

```diff
@@ -185,3 +185,17 @@ def check_synchronization_status(self):
                 sync_status = 'sync_stopped'
         res['google_calendar'] = sync_status
         return res
+
+    def _has_any_active_synchronization(self):
+        """
+        Check if synchronization is active for Google Calendar.
+        This function retrieves the synchronization status from the user's environment
+        and checks if the Google Calendar synchronization is active.
+
+        Returns:
+            bool: True if synchronization is active for Google Calendar, False otherwise.
+        """
+        sync_status = self.env.user.check_synchronization_status()
+        if 'google_calendar' in sync_status and sync_status['google_calendar'] == 'sync_active' and self.check_calendar_credentials():
+            return True
+        return super()._has_any_active_synchronization()
```

Comment:
The super() function must always be executed.
We can use .get() for getting the element from the dict, if it is not present it will be `None` :handshake:
The function `check_synchronization_status` already checks the calendar credentials, it shouldn't be called again.

```suggestion
        res = super()._has_any_active_synchronization()
        if sync_status.get('google_calendar') == 'sync_active':
            res |= True
        return res
```